lib/api/controllers/baseController.js

```diff
@@ -67,137 +69,427 @@ class BaseController {
     return Bluebird.resolve();
   }
 
+  /**
+   * Gets a parameter from a request body and checks that it is a boolean
+   * Contrary to other parameter types, an unset boolean does not trigger an
+   * error, instead it's considered as 'false'
+   *
+   * @param  {Request} request
+   * @param  {string} name - parameter name
+   * @return {boolean}
+   */
+  getBodyBoolean (request, name) {
+    const body = request.input.body;
+
+    if (body === null) {
+      return false;
+    }
+
+    return this._getBoolean(
+      request.input.body,
+      request.context.connection.protocol,
+      name,
+      `body.${name}`);
+  }
 
   /**
-   * Get a boolean param from request input
-   * For HTTP, flag presence mean true value
+   * Gets a parameter from a request body and checks that it is a number
    *
-   * @param {Request} request
-   * @param {string} flagPath
+   * @param  {Request} request
+   * @param  {string} name - parameter name
+   * @param  {number} def - default value to return if the parameter is not set
+   * @return {number}
+   * @throws {api.assert.body_required} If no default value provided and no
+   *                                    request body set
+   * @throws {api.assert.missing_argument} If parameter not found and no default
+   *                                       value provided
+   * @throws {api.assert.invalid_type} If the fetched parameter is not a number
    */
-  tryGetBoolean (request, flagPath) {
-    const
-      flagName = flagPath.split('.').slice(-1),
-      flagValue = _.get(request, `input.${flagPath}`);
-
-    // In HTTP, booleans are flags: if it's in the querystring, it's set, whatever
-    // its value.
-    // If a user needs to unset the option, they need to remove it from the querystring.
-    if ( request.context.connection.protocol !== 'http'
-      && !_.isNil(flagValue)
-      && typeof flagValue !== 'boolean'
-    ) {
-      errorsManager.throw('invalid_value_type', flagName, flagValue);
+  getBodyNumber (request, name, def = null) {
+    const body = request.input.body;
+
+    if (body === null) {
```

Comment:
Why not use `getBody` method here and in the other methods ?

Reply:
For the same reason we made separate methods for these getters: to use `getBody`, we'd need to use it without a default value, otherwise it will return different types, triggering a de-opt, cascaded to callers.

So, we have a choice between (current):

```js
const body = request.input.body;

    if (body === null) {
      if (def !== null) {
        return def;
      }

      assertionError.throw('body_required');
    }
```

And:

```js
let body;

try {
  body = getBody(request);
}
catch (e) {
  if (def !== null) {
    return def;
  }

  throw e;
}
```

Which is longer and more complicated.